lib/bedrock.js

```diff
@@ -213,22 +214,22 @@ api.runOnce = function(id, fn, options, callback) {
   options = options || {};
 
   // listen to run function once
-  process.on('message', runOnce);
+  process.addListener(messageEvent, runOnce);
 
   const ran = false;
   function runOnce(msg) {
     // ignore other messages
     if(!_isMessageType(msg, 'bedrock.runOnce') || msg.id !== id) {
       return;
     }
-
-    process.removeListener('bedrock.runOnce', runOnce);
```

Comment:
Can you explain why the event listener should not be removed after the first matching message arrives?

Reply:
first matching message is neither done or has an error. there appears to be a message sent with the matching name before done or error is set.

Reply:
ok logged some messages:

```
msg { type: 'bedrock.runOnce',
  id: 'bedrock-views.watch',
  done: false }
msg { type: 'bedrock.runOnce',
  id: 'bedrock-views.watch',
  done: true }
msg { type: 'bedrock.runOnce',
  id: 'bedrock-views.watch',
  done: true }
msg { type: 'bedrock.runOnce',
  id: 'bedrock-mongodb.init',
  done: false } 
```

these are from master hence the duplicates.
@mattcollier are you seeing this in your test harness?

Reply:
That message is sent when this worker is selected to run the work. When it completes the work it calls the callback. It should not ever receive any other messages. In fact, if it does, that would be evidence of a bug. So, we should always remove that listener.

Reply:
and for the async stuff

```
msg { type: 'bedrock.runOnceAsync',
  id: 'fpga-poc-node.addAccounts',
  done: false }
runOnceAsync msg { type: 'bedrock.runOnceAsync',
  id: 'fpga-poc-node.addAccounts',
  done: true }

runOnceAsync msg { type: 'bedrock.runOnceAsync',
  id: 'fpga-poc-node.addAccounts',
  done: true,
  error: null }
```

Reply:
My understanding is that the protocol here is to add a listener to listen for exactly one message that matches the message ID (`msg.id`) and then to stop listening.

Reply:
@aljones15, I can't tell from that output if the same worker is receiving multiple messages with the same `msg.id`.

Reply:
@aljones15 I think you'll want to add `process.pid` to your logging.

Reply:
ok so:

https://github.com/digitalbazaar/bedrock/blob/666bb6a7239f182cce928781a4f1781eeca54a56/lib/bedrock.js#L699

and

https://github.com/digitalbazaar/bedrock/blob/666bb6a7239f182cce928781a4f1781eeca54a56/lib/bedrock.js#L750

that result in the listener being removed, then the function runs once, then the results are not sent back.